lib/codegen/src/flowgraph.rs

```diff
@@ -31,7 +31,25 @@ use std::mem;
 use timing;
 
 /// A basic block denoted by its enclosing Ebb and last instruction.
-pub type BasicBlock = (Ebb, Inst);
+pub struct BasicBlock {
+    /// Enclosing Ebb key.
+    pub ebb: Ebb,
+    /// Last instruction in the basic block.
+    pub inst: Inst,
+}
+
+impl BasicBlock {
+    /// Convenient method to construct new BasicBlock.
+    pub fn new(ebb: Ebb, inst: Inst) -> Self {
+        Self { ebb, inst }
+    }
+}
+
+impl PartialEq for BasicBlock {
+    fn eq(&self, rhs: &Self) -> bool {
+        self.ebb == rhs.ebb && self.inst == rhs.inst
+    }
+}
```

Comment:
Instead of manally implementing `eq`, we can just add `#[derive(PartialEq, Eq)]` to the `BasicBlock` struct and we'll get an `eq` function automatically :-).